addons/mail/models/mail_tracking_duration_mixin.py

```diff
@@ -110,3 +118,114 @@ def _get_duration_from_tracking(self, trackings):
             previous_date = tracking['create_date']
 
         return json
+
+    """
+    Rotting logic
+
+    The rotting feature enables resources to mark themselves as stale if enough time has passed since they were last updated
+    by an user.
+    To enable this behavior, the following must be done:
+    - The model pointed to by _track_duration_field must have a "Days to rot" integer field, representing the number of days before
+    the resource is considered stale. That field must be identified on the inheriting model by setting _stage_day_rot_field.
+    - Several methods must be extended/overriden:
+
+      - _compute_date_rot() must be overriden to update its @depends to trigger if the "days to rot field is updated:
+            @api.depends('stage_id.day_rot')
+            def _compute_date_rot(self):
+                return super()._compute_date_rot()
+      - _resource_is_not_rotting_hook(task) must be overriden to add additional conditions for which a stage IS NOT rotting
+        (e.g. a task that has been closed):
+            def _resource_is_not_rotting_hook(self, task):
+                if task.is_closed:
+                    return True
+                return super()._resource_is_not_rotting_hook(task)
+      - _search_is_rotting() must be overriden to update the returned domain with additional conditions for which a stage COULD BE rotting
+        (and condition, all need to be true):
+            def _search_is_rotting(self, operator, value):
+                sup = super()._search_is_rotting(operator, value)
+                return Domain.AND([sup, [('is_closed', '=', True)]])
+
+    - The date_rot, is_rotting, day_rotting, last_activity fields need to be added to the relevant views
+        (as well as the "days to rot" field on the tracking model).
+        You may want to use the rotting_form and rotting_kanban field widgets to display the fields visually on form and kanban view
+        (please note- these widgets need both day_rotting and is_rotting fields on the view to function).
+
+
+    Note that if _stage_day_rot_field is not set, or if the value stored by the field pointed by _stage_day_rot_field is 0,
+    then the resource will never rot.
+    """
+
+    date_rot = fields.Date('Date on which this resource will start rotting', compute="_compute_date_rot", store=True)
+    is_rotting = fields.Boolean('Rotting', compute='_compute_rotting', search='_search_is_rotting')
+    day_rotting = fields.Integer('Days Rotting', help='Day count since this resource was last updated',
+        compute='_compute_rotting')
+    last_activity = fields.Date('Date of last activity', compute="_compute_last_activity", store=True, readonly=False)
+
+    @api.depends('write_date')
+    def _compute_last_activity(self):
+        for resource in self:
+            resource.last_activity = self.env.cr.now()
+
+    def _get_day_rot_value(self):
+        self.ensure_one()
```

Comment:
Docstring for what to expect in return? / Maybe rename `get_day_count_to_rotting` so it's clear we're expecting a number